src/test/java/com/razorpay/InvoiceClientTest.java

```diff
@@ -0,0 +1,140 @@
+        try {
+            mockResponseFromExternalClient(json);
+            mockResponseHTTPCodeFromExternalClient(200);
+            List<Invoice> fetch = invoiceClient.fetchAll();
+            assertNotNull(fetch);
+            assertEquals(true,fetch.get(0).has("type"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void fetch() throws RazorpayException {
+
+        String json = "{\n  \"issued_at\": 1481541533,\n  \"customer_details\":\n  {\n    \"customer_name\": \"Gaurav Kumar\",\n    \"customer_email\": \"gaurav.kumar@example.com\",\n    \"customer_contact\": \"9999999999\"\n  },\n  \"short_url\": \"http://bit.ly/link\", \n  \"receipt\": null, \n  \"entity\": \"invoice\", \n  \"currency\": \"INR\", \n  \"paid_at\": null, \n  \"view_less\": true, \n  \"id\": \"inv_DAweOiQ7amIUVd\", \n  \"customer_id\": \"cust_E7q0trFqXgExmT\", \n  \"type\": null, \n  \"status\": \"issued\", \n  \"description\": \"random decsription\", \n  \"order_id\": \"order_random_id\", \n  \"sms_status\": \"pending\", \n  \"date\": 1481541533, \n  \"payment_id\": null, \n  \"amount\": 100,\n  \"email_status\": \"pending\",\n  \"created_at\": 1481541534\n}";
+        try {
+            mockResponseFromExternalClient(json);
+            mockResponseHTTPCodeFromExternalClient(200);
+            Invoice fetch = invoiceClient.fetch(INVOICE_ID);
+            assertNotNull(fetch);
+            assertEquals(INVOICE_ID,fetch.get("id"));
```

Comment:
try to assert multiple things, see if response does not get altered in processing, test should tell if there is processing mistake

Reply:
ok i will add multiple assert for testing